### PR TITLE
feat(validate): refactor validate module, allow dynamic rules

### LINKED ISSUE #35: Make validate module great again
Currently, `stapp-validate` accepts only a static set of validation rules. Sometimes rules have to be created dynamically, and providing an application's state to the rule itself doesn't solve the issue.

**Solution**
`rules` field of a configuration object should accept a function:
```diff
type ValidateConfig<State> = {
+  rules: ValidationRules<State> | (state: State) => ValidationRules<State>
-  rules: ValidationRules<State>
  validateOnInit?: boolean
  setTouchedOnSubmit?: boolean
}
```

### PR DESCRIPTION
**Pull request by dmitry-korolev**
closes #35

### CODE REVIEW

File: packages/stapp-rxjs/src/combineEpics/combineEpics.ts [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
+import { from as RxFrom, Observable as rxObservable } from 'rxjs'
 
 // Models
-import { Epic } from 'stapp'
+import { Dispatch, Epic as StappEpic, Event } from 'stapp'
 
-const fromOrEmpty = <T>(stream: Subscribable<T> | void): Observable<T> => {
-  return Observable.from(stream || EMPTY)
+export type Epic<State> = (
+  event$: rxObservable<Event<any, any>>,
```
[the remark this answers is not in the document]
Comment (dmitry-korolev) [reply packages/stapp-rxjs/src/combineEpics/combineEpics.ts]: Epic may emit any Event.

File: packages/stapp-rxjs/src/combineEpics/combineEpics.ts [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+    dispatch: Dispatch<State>
+    getState(): State
+  }
+) => rxObservable<any> | void
+
+const fromOrEmpty = <T>(stream: rxObservable<T> | void): Observable<T> => {
+  if (!stream) {
+    return EMPTY
+  }
+  return Observable.from(stream as any)
```
[the remark this answers is not in the document]
Comment (dmitry-korolev) [reply packages/stapp-rxjs/src/combineEpics/combineEpics.ts]: I'll have to fine-tune light-observable typings later. Currently types are not compatible.

File: packages/stapp-rxjs/src/combineEpics/combineEpics.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vkhv) [reply packages/stapp-rxjs/src/combineEpics/combineEpics.ts]: ok

File: packages/stapp-rxjs/src/combineEpics/combineEpics.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (vkhv) [reply packages/stapp-rxjs/src/combineEpics/combineEpics.ts]: ok)

**Pull request merged**